app/components/UserValidator/index.tsx

```diff
@@ -15,16 +15,28 @@ import type { Required } from 'utility-types';
 
 type UserWithUsername = Required<Partial<UserSearchResult>, 'username'>;
 
-type Props = {
+type Res = {
+  payload: unknown;
+};
+
+type Props = Omit<
+  ComponentProps<typeof SearchPage<UserSearchResult>>,
+  'handleSelect'
+> & {
   clearSearch: () => void;
-  handleSelect: (arg0: UserWithUsername) => Promise<User>;
+  handleSelect: (arg0: UserWithUsername) => Promise<User | Res>;
```

Comment:
Couldn't name this something that made more sense? Like `unknownResponse` or something

Reply:
Yes